frontend/ui/elements/mass_storage.lua

```diff
@@ -54,15 +62,19 @@ function MassStorage:start(never_ask)
 
     if not never_ask and self:requireConfirmation() then
         local ConfirmBox = require("ui/widget/confirmbox")
-        UIManager:show(ConfirmBox:new{
+        UIManager:show(ConfirmBox:new {
             text = _("Share storage via USB?"),
             ok_text = _("Share"),
             ok_callback = function()
+                -- save document settings before activating USBMS:
+                saveDocSettings()
```

Comment:
I think you could just do here (and below) (and remove your function above):
`UIManager:flushSettings()`
which will also have global settings saved when in file manager.

Reply:
You're right! Thanks. I'll modify my PR accordingly.